tfx/experimental/pipeline_testing/dummy_launcher_test.py

```diff
@@ -0,0 +1,126 @@
+        name='FakeComponent',
+        input_channel=channel_utils.as_channel([input_artifact]),
+        output_channel=channel_utils.as_channel([output_artifact]))
+    self.driver_args = data_types.DriverArgs(enable_cache=True)
+
+    self.pipeline_info = data_types.PipelineInfo(
+        pipeline_name='Test', pipeline_root=self.pipeline_root, run_id='123')
+
+  @mock.patch.object(publisher, 'Publisher')
+  def testCustomDummyExecutor(self, mock_publisher):
+    # verify whether custom dummy substitution works
+    mock_publisher.return_value.publish_execution.return_value = {}
+    tf.io.gfile.makedirs(self.record_path)
+    component_ids = []
+    component_map = \
+        {'_FakeComponent.FakeComponent': dummy_executor.CustomDummyExecutor}
+
+    MyDummyLauncher = \
+        dummy_component_launcher.create_dummy_launcher_class(
+            self.record_path,
+            component_ids,
+            component_map)
+
+    launcher = MyDummyLauncher.create(
+        component=self.component,
+        pipeline_info=self.pipeline_info,
+        driver_args=self.driver_args,
+        metadata_connection=self.metadata_connection,
+        beam_pipeline_args=[],
+        additional_pipeline_args={})
+    launcher.launch()
+
+    output_path = self.component.outputs['output'].get()[0].uri
+    generated_file = os.path.join(output_path, "test.txt")
+    self.assertTrue(tf.io.gfile.exists(generated_file))
+    contents = file_io.read_file_to_string(generated_file)
+    self.assertEqual('custom component', contents)
+
+  @mock.patch.object(publisher, 'Publisher')
+  def testDummyExecutor(self, mock_publisher):
+    # verify whether dummy substitution works
+    mock_publisher.return_value.publish_execution.return_value = {}
+
+    record_file = os.path.join(self.record_path, 'output/recorded.txt')
+    tf.io.gfile.makedirs(os.path.dirname(record_file))
+    file_io.write_string_to_file(record_file, "hello world")
+    print("record_file")
+
+    component_ids = ['_FakeComponent.FakeComponent']
+    component_map = {}
+
+    MyDummyLauncher = \
+        dummy_component_launcher.create_dummy_launcher_class(self.record_path,
+                                                             component_ids,
+                                                             component_map)
+
+    launcher = MyDummyLauncher.create(
+        component=self.component,
+        pipeline_info=self.pipeline_info,
+        driver_args=self.driver_args,
+        metadata_connection=self.metadata_connection,
+        beam_pipeline_args=[],
+        additional_pipeline_args={})
+    launcher.launch()
+
+    output_path = self.component.outputs['output'].get()[0].uri
+    copied_file = os.path.join(output_path, "recorded.txt")
+    self.assertTrue(tf.io.gfile.exists(copied_file))
+    contents = file_io.read_file_to_string(copied_file)
+    self.assertEqual('hello world', contents)
+
+if __name__ == '__main__':
```

Comment:
Please add one more test case:
Check if dummy launcher executes the original executor, when not having dummy executors.